src/games/strategy/triplea/ai/proAI/ProCombatMoveAI.java

```diff
@@ -967,12 +946,12 @@ private void determineUnitsToAttackWith(final Map<Territory, ProTerritory> attac
         final int isCanHold = canHold ? 1 : 0;
         final int isCantHoldAmphib = !canHold && !patd.getAmphibAttackMap().isEmpty() ? 1 : 0;
         final int isFactory = ProMatches.territoryHasInfraFactoryAndIsLand(player).match(t) ? 1 : 0;
-        final int isFFA = utils.isFFA(data, player) ? 1 : 0;
+        final int isFFA = ProUtils.isFFA(data, player) ? 1 : 0;
```

Comment:
I see, then there are other problems, specifically long method code smell and mixing of levels of abstraction. The chunk of code from lines 944 to 956 could live nicely in a private method called "findAttackValue", similar for the next block "determineWhetherToRemoveAttack", and notably all of the main block level comments. 

This is much easier to understand:

```
public void foo(Territory t) {
   int attackValue= findAttackValue(t);
   boolean doAttack = determienWhetherToRemoveAttack(attackValue);
  :
 :
} 
```

That compared to one single long method where "sections" are delimited by comment headers. Having one method call a series of private methods allows for that one method to stay at the same logical level of abstraction.

This is often described in terms of games and network packets and for hyperbole, I'l use this example: It's like in this method at one layer we have something that is like "should Germany retreat from the eastern front?" , then the next few lines of code are figuring out how to the mercury on a thermometer.